lib/Component.lua

```diff
@@ -182,6 +188,8 @@ function Component:_forceUpdate(newProps, newState)
 		)
 	end
 
+	self._canSetState = true
```

Comment:
An interesting implication here: the state of your parent can't have its state updated while you're in the middle of updating. Calling a function in your `didUpdate` callback that indirectly invokes `setState` on a parent component would thus fail.

I wonder if that's a case that we hit in real code, and if that's a bug!

Reply:
This will only throw an error if this setState indirection happens synchronously. If you spawn another thread and trigger the parent `setState` call there, or do it from an event connection, there won't be an error.

Since most use-cases I can see for calling `setState` in `didUpdate` involve some sort of blocking operation (that stalls out the entire render until it's complete if it's done synchronously anyway), I don't think this is a major issue.

Hypothetically this could be worked around by deferring `didUpdate` invocations until after the entire tree has re-rendered, parents and all, but I don't think this is worth the effort.